kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,280 @@
+    "
+
+    Write-Host $usage
+}
+
+function ReadKubeclusterConfig($ConfigFile)
+{
+    # Read the configuration and initialize default values if not found
+    $Global:ClusterConfiguration = ConvertFrom-Json ((GetFileContent $ConfigFile -ErrorAction Stop) | out-string)
+    if (!$Global:ClusterConfiguration.Install)
+    {
+        $Global:ClusterConfiguration | Add-Member -MemberType NoteProperty -Name Install -Value @{ 
+            Destination = "$env:ALLUSERSPROFILE\Kubernetes";
+        }
+    }
+
+    if (!$Global:ClusterConfiguration.Kubernetes)
+    {
+        throw "Kubernetes information missing in the configuration file"
+    }
+    if (!$Global:ClusterConfiguration.Kubernetes.Source)
+    {
+        $Global:ClusterConfiguration.Kubernetes | Add-Member -MemberType NoteProperty -Name Source -Value @{
+            Release = "1.15.0";
+        }
+    }
+    if (!$Global:ClusterConfiguration.Kubernetes.Master)
+    {
+        throw "Master information missing in the configuration file"
+    }
+
+    if (!$Global:ClusterConfiguration.Kubernetes.Network)
+    {
+        $Global:ClusterConfiguration.Kubernetes | Add-Member -MemberType NoteProperty -Name Network -Value @{
+            ServiceCidr = "10.96.0.0/12";
+            ClusterCidr = "10.244.0.0/16";
+        }
+    }
+
+    if (!$Global:ClusterConfiguration.Cni)
+    {
+        $Global:ClusterConfiguration | Add-Member -MemberType NoteProperty -Name Cni -Value @{
+            Name = "flannel";
+            Plugin = @{
+                Name = "vxlan";
+            };
+            InterfaceName = "Ethernet";
+        }
+    }
+
+    if ($Global:ClusterConfiguration.Cni.Plugin.Name -eq "vxlan")
+    {
+        if (!$Global:ClusterConfiguration.Kubernetes.KubeProxy)
+        {
+            $Global:ClusterConfiguration.Kubernetes | Add-Member -MemberType NoteProperty -Name KubeProxy -Value @{
+                    Gates = "WinOverlay=true";
+            }
+        }
+    }
+
+    if (!$Global:ClusterConfiguration.Cri)
+    {
+        $Global:ClusterConfiguration | Add-Member -MemberType NoteProperty -Name Cri -Value @{
+            Name = "dockerd";
+            Images = @{
+                Pause = "mcr.microsoft.com/k8s/core/pause:1.0.0";
+                Nanoserver = "mcr.microsoft.com/windows/nanoserver:1809";
+                ServerCore = "mcr.microsoft.com/windows/servercore:ltsc2019";
+            }
+        }
+    }
+}
+function LoadPsm1($Path)
+{
+    $tmpPath = [io.Path]::Combine([System.IO.Path]::GetTempPath(), [io.path]::GetFileName($Path))
+    wget $Path -o $tmpPath
+    ipmo $tmpPath  -DisableNameChecking
+    Remove-Item $tmpPath
+}
+###############################################################################################
+# Download pre-req scripts
+
+$helperPath = "https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/kubeadm/v1.15.0/KubeClusterHelper.psm1"
```

Comment:
We have https://sigs.k8s.io, but I don't think you can link to a raw file that way. I would think that updating aliases would cause problems, I'd rather version these things together and force users to update the scripts together. We probably shouldn't even be reaching out to the internet for these things IMHO.